src/kvstore/raftex/RaftPart.cpp

```diff
@@ -1113,22 +1124,16 @@ bool RaftPart::leaderElection() {
                         << host->idStr();
                 return via(
                     eb,
-                    [&voteReq, &host] ()
+                    [&voteReq, &host, eb] ()
                             -> Future<cpp2::AskForVoteResponse> {
-                        return host->askForVote(voteReq);
+                        return host->askForVote(voteReq, eb);
                     });
             })
             | gen::as<std::vector>(),
             // Number of succeeded required
             quorum_,
             // Result evaluator
-            [hosts, this](size_t idx, cpp2::AskForVoteResponse& resp) {
-                if (resp.get_error_code() == cpp2::ErrorCode::E_LOG_STALE) {
-                    LOG(INFO) << idStr_ << "My last log id is less than " << hosts[idx]->address()
-                              << ", double my election interval.";
-                    uint64_t curWeight = weight_.load();
-                    weight_.store(curWeight * 2);
-                }
+            [hosts] (size_t idx, cpp2::AskForVoteResponse& resp) {
```

Comment:
After modify the collectN, Is there any difference with the original logic?

Reply:
Actually no difference, but we can handle the corresponding ErrorCode, e.g. we can handle `E_EXCEPTION` to step down later.

Reply:
make sense.